### ISSUE TITLE
Last update breaks mask behavior with Vuetify (only on Firefox?)

### ISSUE THREAD

**Issue by cll0ud** [RonaldJerez/vue-input-facade#17]
**Describe the bug**
After upgrading from 1.2 to 1.3 the mask doesn't work correctly in Vuetify's inputs. After typing a character that shouldn't be allowed it gets added to the value and the mask breaks.

**To Reproduce**
Just add the directive to the input.
`<v-text-field
                v-facade="'AAAA####'"
                counter="8"
                label="AAAA####"
              />`

**Screenshots**
I've got two gifs showing how it worked on 1.2 and the problem after upgrading to 1.3:
[Version 1.2 (Working)](https://imgur.com/YuoFJvD)
[Version 1.3 (Nor Working)](https://imgur.com/CZLYjml)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Firefox
 - Version: 77.0.1

**Additional context**
I've tried on Firefox, Chrome and Edge and it seems that this only happens on Firefox. Not sure about other Browsers. I had to downgrade to 1.2 for now as I need this feature to be working on Firefox.

**Comment by RonaldJerez** [comment]
Interesting, I'll take a look at whats happening.  There was some vuetify specific code to fix the issue of the initial value not been masked with vuetify so that could be affecting it.

**Comment by RonaldJerez** [comment]
This has been fixed now.  When I made the change to fix #11 I noticed that vuetify is emitting input twice at load.  I modified my emit logic to try to avoid also emitting from the directive twice.  For some reason however the 2nd emit is needed in Firefox.  

So as a side effect of fixing this, you will notice two input events at init when using vuetify and apparently in Firefox also getting two inputs while typing.

**Comment by cll0ud** [comment]
Working fine now, thank you!

**Comment by deraw** [comment]
Hi @RonaldJerez, it seems that there is still a bug present on Firefox when accessing the unmasked value with the directive:

```vue
<VTextField
    v-facade="'####'"
    label="####"
    @input.native="event = $event"
/>

<span v-if="event && event.target">
	{{ event.target.unmaskedValue }}
</span>
```

It works fine on Chrome and Edge, but on Firefox I can enter letters when I souldn't be able to, but they don't get into the `unmaskedValue`

Do you want me to open a new issue?

**Comment by tskimmett** [comment]
@RonaldJerez 

I encountered the same problem that I think @deraw is referring to. After doing some digging and experimenting, I've come to the conclusion that Firefox has a bug (or at least an inconsistency) in their implementation of event handling.

The vue-input-facade directive adds a _capturing_ event listener to the input, which theoretically helps ensure that the masking code executes _before_ normal vue event handlers (which do not specify `useCapture: true`). Chrome and Edge correctly execute the directive's `input` handler first, but Firefox does not.

Simple demonstration of the issue:

`Edge (chromium)`
![image](https://user-images.githubusercontent.com/4603206/115619693-34e96c00-a2c2-11eb-8f99-9511bf7212f7.png)

`Firefox`
![image](https://user-images.githubusercontent.com/4603206/115619760-4b8fc300-a2c2-11eb-9054-5e2d0253d265.png)

I haven't looked too much further into this yet, but I did make a quick change to the directive to add the `input` event handler on the input element's parent, and the issue in Firefox went away.

Basic explanation of capture/bubble event phases: https://www.quirksmode.org/js/events_order.html
